src/clients/ClientDetails.tsx

```diff
@@ -194,9 +197,32 @@ export default function ClientDetails() {
   });
 
   const [client, setClient] = useState<ClientRepresentation>();
+  const [clients, setClients] = useState<ClientRepresentation[]>([]);
+  const [clientRoles, setClientRoles] = useState<RoleRepresentation[]>([]);
+  const [users, setUsers] = useState<UserRepresentation[]>([]);
+
+  useFetch(
+    async () => {
+      const clients = await adminClient.clients.find();
+
+      const roles = await adminClient.roles.find();
+
+      const users = await adminClient.users.find({ realm });
+
+      return { clients, roles, users };
+    },
+    ({ clients, roles, users }) => {
+      setClients(clients);
+      setClientRoles(roles);
+      setUsers(users);
+    },
```

Comment:
It's better to do these in parallel like this:
```suggestion
    () =>
      Promise.all([
        adminClient.clients.find(),
        adminClient.roles.find(),
        adminClient.users.find(),
      ]),
    ([clients, roles, users]) => {
      setClients(clients);
      setClientRoles(roles);
      setUsers(users);
    },
```